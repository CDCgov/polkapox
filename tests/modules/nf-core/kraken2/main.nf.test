nextflow_process {

    name "Test Process KRAKEN2"
    script "../../../../modules/nf-core/kraken2/main.nf"
    process "KRAKEN2"

    tag "kraken2"
    tag "paired"
    tag "classified"

    test("kraken2 paired-end, classified only") {
        when {
            process {
                """
                input[0] = Channel.of([
                    [ id: 'test_sub', single_end: false ],
                    [
                        file("${projectDir}/tests/nf-test-data/test_sub_1.fq.gz", checkIfExists: true),
                        file("${projectDir}/tests/nf-test-data/test_sub_2.fq.gz", checkIfExists: true)
                    ]
                ])
                input[1] = file("${projectDir}/test_data/orthopox_t2t_kdb", checkIfExists: true)
                input[2] = true   // save_output_fastqs_classified
                input[3] = false  // save_output_fastqs_unclassified
                input[4] = true   // save_reads_assignment
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}

