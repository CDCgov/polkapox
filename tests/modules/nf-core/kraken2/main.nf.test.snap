{
    "kraken2 paired-end, classified only": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "4": [
                    
                ],
                "classified_reads_assignment": [
                    
                ],
                "classified_reads_fastq": [
                    
                ],
                "report": [
                    
                ],
                "unclassified_reads_fastq": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-07-07T14:03:56.857764604"
    },
    "kraken2 stub test (paired-end)": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "4": [
                    
                ],
                "classified_reads_assignment": [
                    
                ],
                "classified_reads_fastq": [
                    
                ],
                "report": [
                    
                ],
                "unclassified_reads_fastq": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-06-25T16:07:28.63312695"
    },
    "kraken2 paired-end, classified + unclassified": {
        "content": [
            [
                
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-06-25T16:07:22.312962204"
    }
}